src/main/kotlin/com/workos/directorysync/DirectorySyncApi.kt

```diff
@@ -32,9 +32,27 @@ class DirectorySyncApi(private val workos: WorkOS) {
    * Fetches the list of directories.
    */
   @JvmOverloads
-  fun listDirectories(paginationParams: PaginationParams? = null): DirectoryList {
```

Comment:
Instead of adding a new param here, let's follow the pattern in

https://github.com/workos-inc/workos-kotlin/blob/main/src/main/kotlin/com/workos/directorysync/DirectorySyncApi.kt

We can create a new options class like ListDirectoryGroupOptions that extend PaginationParams. I don't think this would be a breaking change, as I _believe_ passing in PaginationParams should work, but we'd have to test this.

Reply:
I refactored to add a ListDirectoriesOptions class following the pattern from ListDirectoryGroupOptions that you flagged and it works, but I do notice in the test that I had to update the test for using params only to use the new ListDirectoriesOptions class instead of PaginationParams directly. It still allows for the params to be sent without the organization param, but it threw an error when using PaginationParams. When I updated it to use ListDirectoryGroupOptions instead it was all good. 

Looking at how the example app calls this endpoint, it does seem like it would be a breaking change where people would need to update that call to use the new class if I'm understanding the code correctly. Lmk what you think!

Reply:
Thanks for doing that! Yeah I guess since Java is strongly typed, it's a breaking change unfortunately. Let me consult with @robframpton. I wonder if we want to move all of the methods to use a specific params object and increment the major version build so this won't be an issue for other API changes as well.

Reply:
For sure, let me know what you guys end up deciding!

Reply:
Can we add an overload method that takes in the new options class?

```kotlin
fun listDirectories(paginationParams: PaginationParams? = null): DirectoryList {
  // cast PaginationParams argument to ListDirectoriesOptions type
  return listDirectories(listOptionsCastedFromPaginationParams)
}

fun listDirectories(listDirectoryOptions: ListDirectoriesOptions? = null): DirectoryList {
  // Implementation
}
```

That way we pave a path forward for new functionality but avoid breaking any existing integrations.

Reply:
Actually that won't work since it's the same number of arguments. 🤔

Reply:
Just so I make sure I understand correctly, is what we want to do is to create one class of options params, that extends the PaginationParams class, and then use that one class in all the methods as the input object? Or, just create similar options classes for each method that doesn't already have one?

Reply:
The latter -- create similar options classes for other methods in the *Api classes that could have this issue. Actually, after doing a quick scan, this might be the only one that needs it.

Reply:
I'll work on a PR for it today!